src/OpenTelemetry/Metrics/MetricReaderExt.cs

```diff
@@ -133,20 +133,22 @@ internal List<Metric> AddMetricsListWithViews(Instrument instrument, List<Metric
                     }
 
                     if (this.instrumentIdentityToMetric.TryGetValue(instrumentIdentity, out var existingMetric))
-                    {
-                        metrics.Add(existingMetric);
-                        continue;
-                    }
-
-                    if (this.metricStreamNames.Contains(metricStreamName))
                     {
                         OpenTelemetrySdkEventSource.Log.DuplicateMetricInstrument(
                             metricName,
                             meterName,
                             "Metric instrument has the same name as an existing one but differs by description, unit, or instrument type. Measurements from this instrument will still be exported but may result in conflicts.",
                             "Either change the name of the instrument or use MeterProviderBuilder.AddView to resolve the conflict.");
+
+                        if (metrics.Count == 0)
+                        {
+                            metrics.Add(existingMetric);
+                        }
+
```

Comment:
The else part is where we ignore the view and move on - we should do `OpenTelemetrySdkEventSource.Log.MetricViewIgnored` here.

Reply:
Yep good catch.